src/main/java/com/impactupgrade/nucleus/service/logic/DonationService.java

```diff
@@ -77,6 +77,14 @@ public void createDonation(PaymentGatewayEvent paymentGatewayEvent) throws Excep
             existingDonation.get().id, paymentGatewayEvent.getCrmDonation().transactionId);
         crmService.updateDonation(paymentGatewayEvent.getCrmDonation());
         return;
+
+      } else if (paymentGatewayEvent.getCrmDonation().status == CrmDonation.Status.FAILED
+          && existingDonation.get().status == CrmDonation.Status.FAILED) {
+        // allow updates to error attempts, to catch all failed retries
+        env.logJobInfo("found existing CRM donation {} using transaction {}, in error state; updating it with the reattempt...",
+            existingDonation.get().id, paymentGatewayEvent.getCrmDonation().transactionId);
+        crmService.insertFailedDonationAttempt(paymentGatewayEvent.getCrmDonation());
+        return;
```

Comment:
Hey @VSydor, I'm really sorry for letting this sit for so long, but coming back to it. I'm wondering if we really need this chunk of code and `insertFailedDonationAttempt` added to `CrmService`. DonationService.insertDonation and CrmService.insertDonation already create failed attempts based on the donation status. See SfdcCrmService.setOpportunityFields and its use of the "Failed Attempt" StageName.

What https://app.shortcut.com/impactupgrade/story/12199/differentiate-between-failing-but-retrying-and-no-more-retries-in-stripe-charges is aiming for is to differentiate between the different types of errors and whether or not DR can expect that Stripe is going to continue reattempting the charge.

`From what I've seen in webhooks, there's nothing obvious telling us that a charge is no longer going to be retried. But maybe tapping into the invoice would do it? Or something else?`

What we're instead looking for: is there a way to figure out whether or not a Stripe failed attempt is the only-or-last attempt, or if Stripe plans to try the attempt again x # of days from now? Maybe within the Invoice? Elsewhere?